render-tests/runtime-styling/set-style-layer-reorder/style.json

```diff
@@ -0,0 +1,85 @@
+{
+  "version": 8,
+  "metadata": {
+    "ignored": {
+      "native": "https://github.com/mapbox/mapbox-gl-native/issues/2445"
+    },
+    "test": {
+      "width": 64,
+      "height": 64,
+      "operations": [
+        [
+          "setStyle",
+          {
+            "version": 8,
+            "transition": {
+              "duration": 0
+            },
+            "sources": {
+              "geojson": {
+                "type": "geojson",
+                "data": {
+                  "type": "Point",
+                  "coordinates": [
+                    0,
+                    0
+                  ]
+                }
+              }
+            },
+            "layers": [
+              {
+                "id": "circle-2",
+                "type": "circle",
+                "source": "geojson",
+                "paint": {
+                  "circle-radius": 2,
+                  "circle-color": "red"
+                }
+              },
+              {
+                "id": "circle",
+                "type": "circle",
+                "source": "geojson"
+              }
```

Comment:
Reason that `expected.png` looks right: in the original style, the smaller, red circle (`circle-2`) comes after the black one, so that it appears in the center.

(Note -- in this commit, this test is wrong because the source also changes; fixed in e7e13dad353131388255df54cd619f9311a41702.)